service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ManageHearingsController.java

```diff
@@ -0,0 +1,178 @@
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        if (caseData.getAllocatedJudge() != null) {
+            caseDetails.getData().put("judgeAndLegalAdvisor", setAllocatedJudgeLabel(caseData.getAllocatedJudge()));
+        }
+
+        List<Element<HearingBooking>> hearings = defaultIfNull(caseData.getHearingDetails(), List.of());
+
+        if (hearings.isEmpty()) {
+            caseDetails.getData().put(FIRST_HEARING_FLAG, "Yes");
+        } else {
+            caseDetails.getData().putAll(manageHearingsService.populateInitialFields(caseData));
+        }
+
+        return respond(caseDetails);
+    }
+
+    @PostMapping("/edit-hearing/mid-event")
+    public AboutToStartOrSubmitCallbackResponse populateExistingDraftHearing(
+        @RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        //Only triggered for edit journey
+        if (EDIT_DRAFT.equals(caseData.getUseExistingHearing())) {
+            UUID hearingBookingId = getDynamicListValueCode(caseData.getHearingDateList(), mapper);
+
+            List<Element<HearingBooking>> futureHearings = hearingBookingService.getFutureHearings(
+                caseData.getHearingDetails());
+
+            caseDetails.getData().put(HEARING_DATE_LIST,
+                asDynamicList(futureHearings, hearingBookingId, hearing -> hearing.toLabel(DATE)));
+
+            HearingBooking hearingBooking = manageHearingsService.findHearingBooking(
+                hearingBookingId, caseData.getHearingDetails());
+
+            caseDetails.getData().putAll(manageHearingsService.populateHearingCaseFields(hearingBooking));
+
+            if (hearingBookingId.equals(caseData.getHearingDetails().get(0).getId())) {
+                caseDetails.getData().put(FIRST_HEARING_FLAG, "Yes");
+            }
+        }
+
+        return respond(caseDetails);
+    }
+
+    @PostMapping("/validate-hearing-dates/mid-event")
+    public AboutToStartOrSubmitCallbackResponse validateHearingDatesMidEvent(
+        @RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        List<String> errors = validateGroupService.validateGroup(caseData, HearingDatesGroup.class);
+
+        return respond(caseDetails, errors);
+    }
+
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        HearingBooking hearingBooking = manageHearingsService.buildHearingBooking(caseData);
+
+        if (caseData.getSendNoticeOfHearing() != null && caseData.getSendNoticeOfHearing().equals("Yes")) {
```

Comment:
Thoughts on extracting this to a function in case data or the service?

Reply:
what about
        if ("Yes".equals(caseData.getSendNoticeOfHearing()))
?

Reply:
or that, makes much more sense 🙂